### PR TITLE
Add support for --direct access for cim plugins

### LINKED ISSUE #1418: CIM usage
How to use target-query -f cim  
When I have triage folder or only folder with wbem/repository
If I use path to folder then I have error 
```
target-query APTSimulatorVM_Defender_WBEM/ -f cim
2025-11-29T00:51:58.118031Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/INDEX.BTR>: Can't identify volume system or no volumes found, adding as raw volume instead: <Container type=raw size=5029888 vs=None> [dissect.target.target]
2025-11-29T00:51:58.637985Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/INDEX.BTR>: Can't identify filesystem: <Volume name=None size=5029888 fs=None> [dissect.target.target]
2025-11-29T00:51:58.639163Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/INDEX.BTR>: Failed to find OS plugin, falling back to default [dissect.target.target]
2025-11-29T00:51:58.640287Z [error    ] <Target APTSimulatorVM_Defender_WBEM/INDEX.BTR>: Unsupported plugin for cim: No WBEM repository found [dissect.target.target]
2025-11-29T00:51:58.703832Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/MAPPING1.MAP>: Can't identify volume system or no volumes found, adding as raw volume instead: <Container type=raw size=78460 vs=None> [dissect.target.target]
2025-11-29T00:51:58.726231Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/MAPPING1.MAP>: Can't identify filesystem: <Volume name=None size=78460 fs=None> [dissect.target.target]
2025-11-29T00:51:58.727487Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/MAPPING1.MAP>: Failed to find OS plugin, falling back to default [dissect.target.target]
2025-11-29T00:51:58.728493Z [error    ] <Target APTSimulatorVM_Defender_WBEM/MAPPING1.MAP>: Unsupported plugin for cim: No WBEM repository found [dissect.target.target]
2025-11-29T00:51:58.785934Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/MAPPING2.MAP>: Can't identify volume system or no volumes found, adding as raw volume instead: <Container type=raw size=78460 vs=None> [dissect.target.target]
2025-11-29T00:51:58.806087Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/MAPPING2.MAP>: Can't identify filesystem: <Volume name=None size=78460 fs=None> [dissect.target.target]
2025-11-29T00:51:58.807249Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/MAPPING2.MAP>: Failed to find OS plugin, falling back to default [dissect.target.target]
2025-11-29T00:51:58.808287Z [error    ] <Target APTSimulatorVM_Defender_WBEM/MAPPING2.MAP>: Unsupported plugin for cim: No WBEM repository found [dissect.target.target]
2025-11-29T00:51:58.924121Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/MAPPING3.MAP>: Can't identify volume system or no volumes found, adding as raw volume instead: <Container type=raw size=78460 vs=None> [dissect.target.target]
2025-11-29T00:51:58.957322Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/MAPPING3.MAP>: Can't identify filesystem: <Volume name=None size=78460 fs=None> [dissect.target.target]
2025-11-29T00:51:58.959542Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/MAPPING3.MAP>: Failed to find OS plugin, falling back to default [dissect.target.target]
2025-11-29T00:51:58.961335Z [error    ] <Target APTSimulatorVM_Defender_WBEM/MAPPING3.MAP>: Unsupported plugin for cim: No WBEM repository found [dissect.target.target]
2025-11-29T00:51:59.062442Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/OBJECTS.DATA>: Can't identify volume system or no volumes found, adding as raw volume instead: <Container type=raw size=23937024 vs=None> [dissect.target.target]
2025-11-29T00:51:59.339060Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/OBJECTS.DATA>: Can't identify filesystem: <Volume name=None size=23937024 fs=None> [dissect.target.target]
2025-11-29T00:51:59.342910Z [warning  ] <Target APTSimulatorVM_Defender_WBEM/OBJECTS.DATA>: Failed to find OS plugin, falling back to default [dissect.target.target]
2025-11-29T00:51:59.346074Z [error    ] <Target APTSimulatorVM_Defender_WBEM/OBJECTS.DATA>: Unsupported plugin for cim: No WBEM repository found [dissect.target.target]
```

### PR DESCRIPTION
**Pull request by william-billaud**
use the _get_path api to allows using the plugin in direct access mode.

As the plugins uses multiple files, IMO the more simple ways to uses the plugins is by providing the folder as input.

But this require case insensitive match on files. I'm not sure if this should be the plugins problem (managing case insensitive match) or direct loader vfs should be case insensitive.

closes #1418

### CODE REVIEW

**Comment by william-billaud** [comment]
Another option is to leave the choice up to the user (with an option such as --direct / --insentive-direct), which add more complexity to cli.

**Comment by twiggler** [comment]
Expanding on an idea by @Schamper  to create two virtual filesystems in the direct loader, one case sensitive and one insensitive, perhaps we can augment the `get_paths` interface to take an additional boolean `case_insensitive` to select on the desired filesystem. 

An open question is how to represent the case insensitive filesystem in the target. 
Perhaps it could be a property such as `ifs`.

**Comment by william-billaud** [comment]
So it would be partly the plugins responsibility to choice between a case sensitive or a case insensitive file system? 
Each plugins will have a variable member such as DIRECT_MODE_REQUIRE_CASE_INSENSITIVE, and the get_path would use this variable? This look good to me. 
Would you prefer to implement it in another PR (let me know if you need help) and close this PR after?

File: pyproject.toml [hunk 1]
```diff
@@ -38,7 +38,7 @@ dependencies = [
     "flow.record~=3.21.0",
     "structlog",
 ]
-dynamic = ["version"]
+version = "2.30.1"
```
Comment (twiggler) [inline pyproject.toml]: What is this for?

**Review by twiggler — CHANGES_REQUESTED** [review]

File: tests/loaders/test_direct.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import pytest
+
+from dissect.target import Target
+from tests._utils import absolute_path
+
+
+def test_direct_overlap_warning(
+    tmp_path: Path, caplog: pytest.LogCaptureFixture, monkeypatch: pytest.MonkeyPatch
+) -> None:
+    """Assert direct raise warning in case sensitive mode if some files overlap
```
[the remark this answers is not in the document]
Comment (william-billaud) [reply tests/loaders/test_direct.py]: Problems solved, this was related to a conversion from VFS to string by the direct loader. Thus, the tests iterated over my entire file system, which is quite large.

**Comment by william-billaud** [comment]
@twiggler let me know your position on check_case_insensitive_overlap.

File: dissect/target/loaders/direct.py [hunk 3]
```diff
@@ -29,3 +42,31 @@ def map(self, target: Target) -> None:
...
+            return
+        for f in base_path.iterdir():
+            if f.is_dir():
+                yield from self.yield_all_file_recursively(f, max_depth=max_depth - 1)
+            else:
+                yield f
+
+    def check_case_insensitive_overlap(self) -> bool:
+        """Verify if two differents files will have the same path in a case-insensitive fs"""
+        all_files_list = list(
```
[the remark this answers is not in the document]
Comment (twiggler) [reply dissect/target/loaders/direct.py]: I've been looking at the implementation of `yield_all_file_recursively` and `check_case_insensitive
_overlap`. I noticed the comment mentioning that rglob isn't case-sensitive until Python 3.12. While that's a valid point to consider, the actual case-insensitive check in the current code happens when the paths are converted to lowercase. 

My proposed refactoring keeps this exact same logic but simplifies the first step of gathering the files. Instead of using a custom recursive function, we can use rglob to get all the file paths and then perform the same lowercase comparison. This works correctly on any OS, regardless of the filesystem's native case sensitivity, because the check is done in Python after collecting the files.

Here’s how we could refactor it:

```
# No longer needed
# def yield_all_file_recursively(self, base_path: Path, max_depth: int = 7) -> Iterator[Path]:

def check_case_insensitive_overlap(self) -> bool:
    """Verify if two different files will have the same path in a case-insensitive fs."""

    def get_files(path: Path):
        if not path.exists():
            return []
        if path.is_file():
            return [path]
        # Recursively find all files in the directory
        return list(path.rglob("*"))

    # Create a flat list of all file paths from all input directories
    all_paths = chain.from_iterable(get_files(p) for p in self.paths)
    # Filter out directories, keeping only files
    all_files = [p for p in all_paths if p.is_file()]

    # Compare the count of all files with the count of unique, lowercased file paths
    return len({str(p).lower() for p in all_files}) != len(all_files)
```

(Untested)
What do you think of this approach?

[the remark this answers is not in the document]
Comment (william-billaud) [reply dissect/target/loaders/direct.py]: It's close to my initial approach.
unfortunately it does not work on windows with python <3.12 when target is on a case sensitive FS (e.g mounted virtualbox share).

This is an edge case but I think that we should support it (but it's up to you to decide). But i was not able to catch it in test (it maybe related to the cpython implemention). furthermore it would be really hard to test it in CI (would require a case senstive FS on windows test machine. 
my comment is misleading as in fact this is not related to the case_sensitive parameter, but probably to some inner works when adding this option

Here is the output from a windows VM, where z: is virtualbox share with your suggestion and o has the following content


```
❯ tree o
o
├── u
│   ├── a
│   │   └── 1
│   └── A
│       └── 1
└── U
    ├── a
    │   └── 1
    └── A
        └── 1

6 directories, 4 files                                                 
```

With your suggestion : Warning only raised in 3.12

```
PS C:\Users\RaptorSniper\dissect.target>  uv run --python 3.12 --refresh --extra dev --extra full target-query --direct -f example_namespace Z:\o
2026-01-13T16:09:49.617953Z [warning  ] Direct mode used in case insensitive mode, but this will cause files overlap, consider using --direct-sensitive [dissect.target.loaders.direct]
<example/descriptor hostname=None domain=None field_a='namespace_example' field_b='record'>
PS C:\Users\RaptorSniper\dissect.target>  uv run --python 3.10 --refresh --extra dev --extra full target-query --direct -f example_namespace Z:\o
Using CPython 3.10.19
<example/descriptor hostname=None domain=None field_a='namespace_example' field_b='record'>
PS C:\Users\RaptorSniper\dissect.target>  uv run --python 3.11 --refresh --extra dev --extra full target-query --direct -f example_namespace Z:\o
Using CPython 3.11.14
<example/descriptor hostname=None domain=None field_a='namespace_example' field_b='record'>
PS C:\Users\RaptorSniper\dissect.target>        
```

With previous implementation : Warning raised with all version

```
PS C:\Users\RaptorSniper\dissect.target>  uv run --python 3.10 --refresh --extra dev --extra full target-query --direct -f example_namespace Z:\o
Using CPython 3.10.19
2026-01-13T16:18:52.905258Z [warning  ] Direct mode used in case insensitive mode, but this will cause files overlap, consider using --direct-sensitive [dissect.target.loaders.direct]
<example/descriptor hostname=None domain=None field_a='namespace_example' field_b='record'>
PS C:\Users\RaptorSniper\dissect.target>  uv run --python 3.11 --refresh --extra dev --extra full target-query --direct -f example_namespace Z:\o
Using CPython 3.11.14
2026-01-13T16:19:26.408962Z [warning  ] Direct mode used in case insensitive mode, but this will cause files overlap, consider using --direct-sensitive [dissect.target.loaders.direct]
<example/descriptor hostname=None domain=None field_a='namespace_example' field_b='record'>
PS C:\Users\RaptorSniper\dissect.target>  uv run --python 3.12 --refresh --extra dev --extra full target-query --direct -f example_namespace Z:\o
Using CPython 3.12.10 interpreter at: C:\Users\RaptorSniper\AppData\Local\Microsoft\WindowsApps\PythonSoftwareFoundation.Python.3.12_qbz5n2kfra8p0\python.exe
2026-01-13T16:20:40.384049Z [warning  ] Direct mode used in case insensitive mode, but this will cause files overlap, consider using --direct-sensitive [dissect.target.loaders.direct]
<example/descriptor hostname=None domain=None field_a='namespace_example' field_b='record'>
```

File: dissect/target/loaders/direct.py [hunk 4]
```diff
@@ -1,31 +1,69 @@
...
+        all_paths = chain.from_iterable(get_files(p) for p in self.paths)
+        # Filter out directories, keeping only files
+        all_files = [p for p in all_paths if p.is_file()]
+        # Compare the count of all files with the count of unique, lowercased file paths
+
+        return len({str(p).lower() for p in all_files}) != len(all_files)
+
+    def __repr__(self) -> str:
+        """
+        As DirectLoader does not call super().__init__() self.path is not defined, we need to redefine the __repr__ func
```
[the remark this answers is not in the document]
Comment (william-billaud) [reply dissect/target/loaders/direct.py]: I'm not sure.

Loader expect a single path object, direct has a list of path. Furthermore direct make some operation related to the provided path. 
https://github.com/fox-it/dissect.target/blob/4b7bddcb00d2d4ef5f3184ff69ff8794267856b5/dissect/target/loader.py#L68 

Calling super().__init__(path=paths) won't work. We can still call super().__init__(path=paths[0]) but it's look like more like a hack to me.

File: dissect/target/loaders/direct.py [hunk 5]
```diff
@@ -1,31 +1,95 @@
...
+                "Direct mode used in case insensitive mode, but this will cause files overlap, "
+                "consider using --direct-sensitive"
+            )
+        vfs = VirtualFilesystem(case_sensitive=self.case_sensitive)
         for path in self.paths:
             if path.is_file():
-                vfs.map_file(str(path), str(path))
+                vfs.map_file(str(path), path)
             elif path.is_dir():
-                vfs.map_dir(str(path), str(path))
```
Comment (william-billaud) [inline dissect/target/loaders/direct.py]: @twiggler regarding 

> Can you elaborate on why the string conversion was a problem? According to the typings, both LayerFilesystem::map_dir and LayerFilesystem::map_file should be able to take realpath as a str


As the path variable (that containes the VFS)  was converted to string, the vfs object was lost and vfs.map_file tried to map the string '/' (thus my root fs)

File: dissect/target/loaders/direct.py [hunk 6]
```diff
@@ -1,31 +1,93 @@
...
 
         target.filesystems.add(vfs)
         target._os_plugin = DefaultOSPlugin
+
+    def check_case_insensitive_overlap(self) -> bool:
+        """Verify if two differents files will have the same path in a case-insensitive scenario."""
+        if sys.version_info >= (3, 12) or sys.platform != "win32":
+
+            def get_files(path: Path) -> Iterator[Path]:
+                """Return list of all files recursively."""
```
Comment (Schamper) [inline dissect/target/loaders/direct.py]:
```suggestion
```
